src/sage/graphs/generators/random.py

```diff
@@ -1460,6 +1460,123 @@ def RandomTreePowerlaw(n, gamma=3, tries=1000, seed=None):
         return False
 
 
+def RandomKTree(n, k, seed=None):
+    r"""
+    Return a random `k`-tree on `n` nodes numbered `0` through `n-1`.
+
+    ALGORITHM:
+
+    The algorithm first generates a complete graph on `k+1` vertices. Vertices
+    are subsequently generated by randomly choosing one of the existing cliques
+    in the graph, and creating a new clique by replacing one of the vertices in
+    the selected clique with a newly created one.
+
+    INPUT:
+
+    -  ``n`` -- number of vertices in the k-tree
+
+    -  ``k`` -- within a clique each vertex is connected to `k` vertices. `k`
+      also corresponds to the treewidth of the k-tree
+
+    - ``seed`` -- a ``random.Random`` seed or a Python ``int`` for the random
+      number generator (default: ``None``)
+
+    TESTS::
+
+        sage: g=graphs.RandomKTree(50,5)
+        sage: g.size()
+        235
+        sage: g.order()
+        50
+        sage: g.treewidth()
+        5
+
+    EXAMPLES:
+
+    A (random) `k`-tree has tree width `k`::
+
+        sage: G = graphs.RandomKTree(50, 5)
+        sage: G.treewidth()
+        5
+        sage: G.show() # not tested 
+    """    
+    if n < k + 1:
+        raise ValueError("n must be greater than k + 1")
+
+    if seed is not None:
+        set_random_seed(seed)
+
+    from sage.graphs.generators.basic import CompleteGraph
+
+    g = Graph()
```

Comment:
may be you can set the name of the graph like `g = Graph(name=f"Random {k}-tree of order {n}")` ?